### PR TITLE
Add service_template_embedded_terraform_template

### PR DESCRIPTION
**Pull request by agrare**
TODO
- [x] Include the terraform template ID in the ServiceTemplate config_info

Related:
* https://github.com/ManageIQ/manageiq/pull/22999

```
POST /api/service_templates
{
  "name": "Terraform Template",
  "description": "",
  "display": true,
  "service_template_catalog_id": "1",
  "zone_id": "2",
  "currency_id": "",
  "price": "",
  "config_info": {
    "provision": {
      "repository_id":"1",
      "execution_ttl":"",
      "log_output":"on_error",
      "verbosity":"0",
      "extra_vars": {"FOO": {"default": "BAR"}},
      "credential_id": "1",
      "configuration_script_payload_id": "58",
      "new_dialog_name": "Terraform Dialog"
    },
    "retirement": {
      "log_output": "on_error",
      "verbosity": "0"
    }
  },
  "long_description": "",
  "additional_tenant_ids": [],
  "type": "ServiceTemplateTerraformTemplate",
  "prov_type": "generic_terraform_template"
}

```

### CODE REVIEW

File: app/models/service_template_embedded_terraform_template.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class ServiceTemplateEmbeddedTerraformTemplate < ServiceTemplate
+  def self.default_provisioning_entry_point(_service_type)
+    '/Service/Generic/StateMachines/GenericLifecycle/provision'
+  end
+
+  def self.validate_config_info(options)
+    options
```
Comment (jrafanie) [inline app/models/service_template_embedded_terraform_template.rb:7]: the other service templates (orchestration, ansible playbook, awx, etc.) all seem to return the value of the key `:config_info` in options:

```ruby
  def self.validate_config_info(options)
    config_info = options[:config_info]
    unless config_info[:configuration_script_id] || config_info[:configuration]
      raise _('Must provide configuration_script_id or configuration')
    end

    config_info
  end
```

Should we do that here?

Comment (agrare) [reply app/models/service_template_embedded_terraform_template.rb:7]: This is just a placeholder until I understood what the value of options was, service_template_ansible_playbook and service_template_orchestration/service_template_container_template handle this differently so I want to see where the options come from.

Comment (jrafanie) [reply app/models/service_template_embedded_terraform_template.rb:7]: yeah, i honestly have no idea either

**Comment by Fryguy** [comment]
Let's drop the dynamic dialog stuff for this PR, get it merged, then follow up with dynamic dialog generation.

**Comment by agrare** [comment]
@Fryguy this should be good to go, I took out the dynamic dialog stub

**Comment by Fryguy** [comment]
This PR has been effectively backported to `radjabov` via the merge of master into radjabov.

**Pull request merged**